First version of array modification examples and explanation document

HeavyWombat opened the pull request:
Hello @geofffranks, hello @jhunt, like I said in one of the previous pull requests, I _think_ it would be helpful for Spruce beginners to have a more detailed explanation how array modifications work. This would be my initial version with what came to my mind. Since I am not a native English speaker, it would be great to check that from a writing perspective, too. Thanks in advance for your feedback.

geofffranks commented on doc/arrays.md:
```diff
@@ -0,0 +1,107 @@
+To emphasize the new annotation an empty line was added. This is technically not needed, however it increases the readability.
+
+## Operators that work the same for either type of array
+- `(( append ))` - The append operator tells Spruce to place the following content before the first existing entry.
```
I believe the description for append + prepend are swapped. prepend goes in front of the first entry, append goes after?

geofffranks commented on doc/arrays.md:
```diff
@@ -0,0 +1,107 @@
+
+- `(( replace ))` - The replace operator tells Spruce to completely remove the existing array and replace it with the following entries.
+
+- `(( inline ))` - The inline operator might be a bit confusing at first glance. It takes the given entries and puts them on top of the existing ones based on their respective position in the array (index). The first one of the new entries is merged with the first one of the existing array and so on and so forth. This works with both types of arrays, however it is not _recommended_ to be used wit ...
```
"creates somehow weird results" - I would change that to "can create unexpected results", since the first reads as if we don't know what's going on that would cause weird results, whereas the second reads as if we know why it's happening, but it might not be readily apparent to the end user.

Also, could this be split onto multiple lines, for file-readability?

geofffranks commented:
Added a couple in-line comments, but otherwise it looks great! Nice job, thank you!

HeavyWombat commented:
Thanks for the feedback. I fixed the things you brought up. Also, I added a couple of line breaks in other sections so that the file might be better diff'able in the future.

geofffranks commented:
Thanks!
